openpgp/read.go

```diff
@@ -461,3 +461,31 @@ func checkArmoredDetachedSignature(keyring KeyRing, signed, signature io.Reader)
 	}
 	return checkDetachedSignature(keyring, signed, body)
 }
+
+// CheckDesignatedRevokers will try to confirm any of designated
+// revocation of entity. For this function to work, revocation
+// issuer's key should be found in keyring, otherwise it will return
+// false, nil if keys are not found (meaning: revocation not
+// confirmed).
+func CheckDesignatedRevokers(keyring KeyRing, entity *Entity) (bool, *Key) {
```

Comment:
Right now I'm calling this from client. So for example, `CheckDetachedSignature` would return a `signer` and then client would call `CheckDesignatedRevokers` with that signer. I'm not sure if it should be called automatically from e.g. `ReadMessage`.